addons/common/persistence/jdbc/src/main/java/org/kie/kogito/persistence/jdbc/JDBCProcessInstances.java

```diff
@@ -45,9 +44,30 @@
 
 public class JDBCProcessInstances implements MutableProcessInstances {
 
-    private static final String VERSION = "version";
+    static final String PAYLOAD = "payload";
+    static final String VERSION = "version";
 
-    private static final String PAYLOAD = "payload";
+    enum DatabaseType {
+        POSTGRES("PostgreSQL", "process_instances"),
+        ORACLE("Oracle", "PROCESS_INSTANCES");
+
+        private final String dbIdentifier;
+        private final String tableNamePattern;
+
+        DatabaseType(final String dbIdentifier, final String tableNamePattern) {
+            this.dbIdentifier = dbIdentifier;
+            this.tableNamePattern = tableNamePattern;
+        }
```

Comment:
What about making `Repository` impl selection part of this enum?
```suggestion
        POSTGRES("PostgreSQL", "process_instances", new PostgresRepository()),
        ORACLE("Oracle", "PROCESS_INSTANCES", new OracleRepository());

        private final String dbIdentifier;
        private final String tableNamePattern;
        private final Repository repository;
        
        DatabaseType(final String dbIdentifier, final String tableNamePattern, Repository repository) {
            this.dbIdentifier = dbIdentifier;
            this.tableNamePattern = tableNamePattern;
            this.repository = repository;
        }
        
        public Repository getRepository() {
            return this.repository;
        }
```